cpp/ql/src/Security/CWE/CWE-313/CleartextSqliteDatabase.ql

```diff
@@ -36,16 +36,29 @@ Field getRecField(Class c) {
   result = getRecField(c.getAField().getUnspecifiedType().stripType())
 }
 
+/**
+ * Holds if `source` is a use of a sensitive expression `sensitive`, or
+ * if `source` is the output argument (with a sensitive name) of a function.
+ */
+predicate isSourceImpl(DataFlow::Node source, SensitiveExpr sensitive) {
+  [source.asExpr(), source.asDefiningArgument()] = sensitive
```

Comment:
I did briefly look at this query earlier, but couldn't figure it out at the time. So to further my understanding: why `source.asDefiningArgument()`? Would `source.asIndirectExpr()` have also worked?

Reply:
`source.asDefiningArgument()` is about data coming out of a function, whereas `source.asIndirectExpr()` would give you the data coming into the function.

I would expect the use node obtained by `source.asIndirectExpr()` to also have outgoing flow to the next use, but it may actually be the case that we don't have this since we don't consider data coming out of functions as [uncertain writes](https://github.com/github/codeql/pull/11658) 🤔.

Reply:
Yeah, I see what you mean. I'll see if there's a problem with use-use flow through arguments (and then delete the `source.asDefiningArgument()` if it turns out there's an underlying dataflow issue).